File: java/src/test/java/api/arch/RoleGroupDemoTest.java
package api.arch;

import model.role.RoleGroupListQueryParam;

import java.util.Map;

public class RoleGroupDemoTest {
    public static void main(String[] args) throws Exception {
        RoleGroupDemo roleGroupDemo = new RoleGroupDemo();
        // 列出角色组
        roleGroupList(roleGroupDemo);
        // 创建角色组
        roleGroupCreate(roleGroupDemo);
        // 更新角色组
        roleGroupUpdate(roleGroupDemo);
        // 删除角色组
        roleGroupDelete(roleGroupDemo);
    }

    private static void roleGroupDelete(RoleGroupDemo roleGroupDemo) throws Exception {
        Map<String, Object> result = roleGroupDemo.roleGroupDelete( 2328);
        System.out.println("roleGroupDelete result\n" + result);
    }

    private static void roleGroupUpdate(RoleGroupDemo roleGroupDemo) throws Exception {
        Map<String, Object> result = roleGroupDemo.roleGroupUpdate("Java-name2", 2328);
        System.out.println("roleGroupUpdate result\n" + result);
    }

    private static void roleGroupCreate(RoleGroupDemo roleGroupDemo) throws Exception {
        Map<String, Object> result = roleGroupDemo.roleGroupCreate("Java-name1");
        System.out.println("roleGroupCreate result\n" + result);
    }

    private static void roleGroupList(RoleGroupDemo roleGroupDemo) throws Exception {
        RoleGroupListQueryParam queryParam = new RoleGroupListQueryParam();
        queryParam.setSkip(0);
        queryParam.setLimit(10);
        queryParam.setHas_internal(true);
        queryParam.setHas_sync(true);
        Map<String, Object> result = roleGroupDemo.roleGroupList(queryParam);
        System.out.println("roleGroupList result\n" + result);
    }
}

File: python/test/api/arch/role_group_demo_test.py
import src.api.arch.role_group_demo as role_group_demo
from src.model.role.role_group_list_query_param import RoleGroupListQueryParam



# 测试 列出角色组
def roleGroupList():
    role_group_list_query_param = RoleGroupListQueryParam(0, 10)
    result = role_group_demo.roleGroupList(role_group_list_query_param)
    print('roleGroupList result:', result)

# 测试 创建角色组
def roleGroupCreate():
    result = role_group_demo.roleGroupCreate('python-name')
    print('roleGroupCreate result:', result)

# 测试更新角色组
def roleGroupUpdate():
    result = role_group_demo.roleGroupUpdate('python-name1',2335)
    print('roleGroupUpdate result:', result)


def roleGroupDelete():
    result = role_group_demo.roleGroupDelete(2335)
    print('roleGroupDelete result:', result)


if __name__ == '__main__':
    roleGroupCreate()
    roleGroupList()
    roleGroupUpdate()
    roleGroupDelete()


